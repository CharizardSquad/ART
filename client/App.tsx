import React, { type ReactElement } from 'react'
import logo from './assets/logo.png'
import './sass/App.scss'
import MainContainer from './containers/MainContainer'
// import GBA frsom './assets/GBA.png'

function App (): ReactElement {
  return (
    <div>
      {/* <img src={GBA} alt="Game Boy Advanced Color" id="background" /> */}
      <div id="app-container">
        <div className="header">
          <div className="center-container">
            <div className="logo-container">
              <img src={logo} alt="shiny charizard logo" id="logo" />
            </div>
            <h3>FireSpin API</h3>
          </div>
        </div>
        <MainContainer />
      </div>
    </div>
  )
}
export default App
