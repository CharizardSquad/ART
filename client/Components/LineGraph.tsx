import React, {
  useState,
  useEffect,
  useContext,
  type ReactElement
} from 'react'
import { Line } from 'react-chartjs-2'
import { Chart, CategoryScale, LinearScale, PointElement, LineElement } from 'chart.js'

Chart.register(
  CategoryScale,
  LinearScale,
  PointElement,
  LineElement
)

function LineChart ({ responseTimes }: { responseTimes: number[] }): ReactElement {
  console.log('resTime in lineGraph:', responseTimes)
  
  const data = {
    labels: responseTimes.map((_, index) => index.toString()),
    datasets: [
      {
        label: 'Responce Time',
        data: responseTimes,
        borderColor: 'rgba(75, 192, 192, 1)',
        borderWidth: 2,
        fill: false
      }
    ]
  }
  return (
    <div className="lineGraph">
      <Line data={data} />
    </div>
  )
}

export default LineChart
