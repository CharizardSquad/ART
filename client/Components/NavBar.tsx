import React, { useState, type ReactElement } from 'react'
import { useNavigate } from 'react-router-dom'
// import Login from './Login'

function NavBar (): ReactElement {
  const navigate = useNavigate()

  //   function GoLogin (): void { navigate('/login') }
  return (
    <div className="navBar"> 
      <button type="button" className="navButton" onClick={(): void => { navigate('/Home') }}>
        Home
      </button>
      <button type="button" className="navButton" onClick={(): void => { navigate('/History') }}>
        History
      </button>
    </div>
  )
}

export default NavBar
