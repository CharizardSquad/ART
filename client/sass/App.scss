@import '_graph';

html{
   background: url('https://wallpaperaccess.com/full/2717776.jpg') no-repeat center center fixed; 
  -webkit-background-size: cover;
  -moz-background-size: cover;
  -o-background-size: cover;
  background-size: cover;
}

body {
  display: flex;
  justify-content: center;
  // flex: 1;
  align-items: center;
    height: 100vh;
  padding: 0;
  margin: 0;
}

h3{
    display: flex;
    justify-content: center;
    color: rgb(117, 25, 25);
    text-shadow: -1px 1px 0 rgb(255, 255, 255), 1px 1px 0 rgb(255, 255, 255), 1px -1px 0 rgb(255, 255, 255), -1px -1px 0 rgb(255, 255, 255);
}

.app {
  // display: flex;
  // height: 100vh;
}

#app-container{
    // position: relative;
    // height: 27%;
    // width: 45%;
    // top: 45vh;
    // left: 50%;
    // transform: translate(-50%, -50%);
    // border-radius: 10px;
    // border: 5px solid rgb(0, 0, 0);
    // padding: 10px;
    // align-content: center;
    // background-color: rgb(242, 145, 48);
    // display: flex;
    // flex-direction: column;
    // // align-items: center;
    position: relative;
    margin-top: 27vh;
    left: 50%;
    transform: translate(-50%, -50%);
    // border-radius: 10px;
    border: 5px solid rgb(0, 0, 0);
    padding: 10px;
    // align-content: center;
    background-color: rgb(242, 145, 48);
    display: flex;
    flex-direction: column;
    // align-items: center;
    width: 46vh;
    height: 30vh;
}



.header{
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 10px;
  width: 100%;
}

.center-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  // margin-top: 20px;
}

.input-API {
  display: flex;
  justify-content: center;
  align-items: center;
  /* box-shadow: 0px 4px 6px rgba(0, 0, 0, 0.1); */
  
}

// #background{
//   object-fit: cover;
//   height: 100%;
//   width: 100%
// }

#paste {
  color: rgb(0, 0, 0);
  background-color: rgb(223, 231, 234);
  border-radius: 25px;
  border: 2px solid white;
}

button:hover{
  
}

.logo-container{
    margin: auto;
    width: 100px;
    height: 100px;
    border-radius: 50%;
    overflow: hidden;
    display: flex;
    justify-content: center;
    align-items: center;
    box-sizing: border-box;
}

.logo-container img{
    max-width: 140%;
    max-height: 140%;
}

.logo-title-container {
  display: flex;
  align-items: center; 
}

.input-API{
    margin-top: 10px;
}

// navBar styles
.navBar {
  display: flex;
  gap: 3px;
  align-items: center;
  margin-left: auto;
  position: absolute;
  top: 0;
  right: 0;
  padding: 10px;
}