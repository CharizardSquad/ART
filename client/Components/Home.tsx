import React, { useState, useEffect, type ReactElement } from 'react'
import { useLocation } from 'react-router-dom'
// import axios from 'axios'
import NavBar from './NavBar'
import LineGraph from './LineGraph'
// import type { LocationState } from '../../types/types'

function Home (): ReactElement {
  const location = useLocation()
  const [input, setInput] = useState('')
  const [callInput, setCallInput] = useState('')
  const [responseTimes, setResponseTimes] = useState<number[]>([])
  const [token, setToken] = useState('')

  useEffect(() => {
    if (location.state.token) {
      setToken(location.state.token)
    }
  }, [location.state])

  const handleCallChange = (e: any): void => {
    setCallInput(e.target.value)
  }

  const handleInputChange = (e: any): void => {
    setInput(e.target.value)
  }
  // const handlePasteClick = async (): Promise<any> => {
  //   try {
  //     const pastedText = await navigator.clipboard.readText()
  //     setInput(pastedText)
  //   } catch (err) {
  //     return 'Unable to paste text'
  //   }
  // }

  const handleGetData = async (): Promise<void> => {
    try {
      // console.log('home-token:', token)
      const response = await fetch('/api/add', {
        method: 'POST',
        credentials: 'include',
        headers: {
          'Content-Type': 'application/json',
          Authorization: token
        },
        body: JSON.stringify({
          url: input,
          calls: callInput
        })
      })

      if (!response.ok) {
        throw new Error('Error fetching data')
      }

      const responseData = await response.json()
      console.log('responseData:', responseData) // in ms?
      setResponseTimes(responseData)
    } catch (err) {
      console.error('Error fetching data', err)
    }
  }
  // console.log('right before render:', responseTimes)
  return (
    <div>
      <div id="home-container">
        <h1 id="homeText">Home</h1>
        <input
          placeholder="insert API here"
          type="text"
          className="input-API"
          value={input}
          onChange={handleInputChange}
        />
        {/* <button type="submit" id="paste" onClick={handlePasteClick}>
          Paste
        </button> */}
        <input
          placeholder="Calls"
          type="text"
          className="call-freq"
          value={callInput}
          onChange={handleCallChange}
        />
        <button type="button" onClick={handleGetData}>
          Fetch Data
        </button>
        {responseTimes.length > 0 && <LineGraph responseTimes={responseTimes} />}
      </div>
      <NavBar />
    </div>
  )
}

export default Home
