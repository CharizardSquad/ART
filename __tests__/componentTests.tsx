import {describe, expect, test} from '@jest/globals';
import NavBar from '../client/Components/NavBar';
import Home from '../client/Components/Home';
import LineGraph from '../client/Components/LineGraph';
import App from '../client/App'
import renderer from 'react-test-renderer'
import * as React from 'react'

const mockUsedNavigate = jest.fn();
const mockUsedLocation = jest.fn();
jest.mock('react-router-dom', () => ({
   ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockUsedNavigate,
  // useLocation: () => mockUsedLocation,
}));

// jest.mock('react-router-dom', () => ({
//   ...jest.requireActual('react-router-dom'),
//  useLocation: () => mockUsedLocation
// }));


describe('Create Snapshots', () => {
  test('Check if NavBar snapshot matches', () => {
    const tree:any =  renderer
    .create(<NavBar/>)
    .toJSON()

    expect(tree).toMatchSnapshot()
  });

  // test('Check if Home snapshot matches', () => {
  //   const test:any =  renderer
  //   .create(<Home/>)
  //   .toJSON()

  //   expect(test).toMatchSnapshot()
  // });
  
});

// describe('Check Rendering', () => {
//   test('If our Line Graph Renders properly', () => {
//     let element;
//     beforeAll(()=> {
      
//     })
//   })
// });