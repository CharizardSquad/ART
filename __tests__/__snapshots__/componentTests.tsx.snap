// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Navbar First Snapshot Test 1`] = `
<div
  className="navBar"
>
  <button
    className="navButton"
    onClick={[Function]}
    type="button"
  >
    SignOut
  </button>
  <button
    className="navButton"
    onClick={[Function]}
    type="button"
  >
    History
  </button>
</div>
`;

exports[`Create Snapshots Check if Home snapshot matches 1`] = `
<div>
  <div
    id="home-container"
  >
    <h1
      id="homeText"
    >
      Home
    </h1>
    <input
      className="input-API"
      disabled={true}
      onChange={[Function]}
      placeholder="insert API here"
      type="text"
      value=""
    />
    <button
      id="paste"
      onClick={[Function]}
      type="submit"
    >
      Paste
    </button>
    <input
      className="call-freq"
      onChange={[Function]}
      placeholder="Calls"
      type="text"
      value=""
    />
    <button
      onClick={[Function]}
      type="button"
    >
      Fetch Data
    </button>
  </div>
  <div
    className="navBar"
  >
    <button
      className="navButton"
      onClick={[Function]}
      type="button"
    >
      SignOut
    </button>
    <button
      className="navButton"
      onClick={[Function]}
      type="button"
    >
      History
    </button>
  </div>
</div>
`;

exports[`Create Snapshots Check if NavBar snapshot matches 1`] = `
<div
  className="navBar"
>
  <button
    className="navButton"
    onClick={[Function]}
    type="button"
  >
    SignOut
  </button>
  <button
    className="navButton"
    onClick={[Function]}
    type="button"
  >
    History
  </button>
</div>
`;

exports[`Create Snapshots Check if NavBar snapshot matches 2`] = `
<div>
  <div
    id="home-container"
  >
    <h1
      id="homeText"
    >
      Home
    </h1>
    <input
      className="input-API"
      disabled={true}
      onChange={[Function]}
      placeholder="insert API here"
      type="text"
      value=""
    />
    <button
      id="paste"
      onClick={[Function]}
      type="submit"
    >
      Paste
    </button>
    <input
      className="call-freq"
      onChange={[Function]}
      placeholder="Calls"
      type="text"
      value=""
    />
    <button
      onClick={[Function]}
      type="button"
    >
      Fetch Data
    </button>
  </div>
  <div
    className="navBar"
  >
    <button
      className="navButton"
      onClick={[Function]}
      type="button"
    >
      SignOut
    </button>
    <button
      className="navButton"
      onClick={[Function]}
      type="button"
    >
      History
    </button>
  </div>
</div>
`;
